### ISSUE TITLE
Queries for long/int64 are losing precision

### ISSUE THREAD

**Issue by mabzd** [Azure/azure-documentdb-dotnet#151]
This LINQ:

`query => query.Where(d => d.LongValue == longValue)`

gets converted to this query:

`{"query": "SELECT * FROM root WHERE (root["LongValue "] = 1.40007951498021E+18)"}`

Both `d.LongValue` and `longValue` are long. Precision is lost, nothing is returned.

**Comment by rnagpal** [comment]
DocumentDB is rooted in JS type system to have an impedance-free mapping to JSON data. 
In JS, there is only support for numbers as IEEE754 format and hence, there is a possibility of loss of precision for large integrals.

**Comment by mabzd** [comment]
I don't get this. When I correct this query by hand and enter in Azure Portal's Query Explorer everything works fine. I can store, query and display large integrals with precision when working via Azure and not when working via .NET DocumentDB API. Besides, this JSON has nothing to do with how JS handles numbers because the query is stored in string. Care to explain?

**Comment by rnagpal** [comment]
Can you let us know what is value for longValue in your LINQ expression? We should be using Json.NET to serialize that and constructing the query. Can you also check if you serialize that value using Newtonsoft you get it same as 1.40007951498021E+18?

**Comment by mabzd** [comment]
```[TestMethod]
public void Test()
{
    Console.WriteLine(JsonConvert.SerializeObject(new {LongValue = 1400079514980214528}));
}
```

Output: `{"LongValue":1400079514980214528}`

**Comment by rnagpal** [comment]
Thanks for the info. This is a bug in out SDK and we are working on a fix for this. We plan to release it early next week. Until then you can use the "SQL query syntax" instead of LINQ.

**Comment by rnagpal** [comment]
@mabzd This should be fixed in our latest SDK release 1.10.0 
Please let us know if that works for you.

**Comment by toddabel** [comment]
Has anyone also seen issues with the Bitwise query not working correctly with larger values?
